src/components/input/input.stories.mdx

```diff
@@ -33,3 +41,64 @@ You can also add `c-input` to `<select>` elements, in which case you’ll get a
     })}
   </Story>
 </Preview>
+
+## Elastic textarea
+
+Add `js-elastic-textarea` to a `<textarea>` element\* to have it expand as the user types more than will fit.
+
+If the `rows` attribute is set, it will be used as the minimum height.
+
+<p>
+  <small>* see also the JavaScript usage notes below</small>
+</p>
+
+<Preview>
+  <Story name="Elastic Textarea" height="350px">
+    {() => {
+      // Use storybook hooks to trigger JS after story renders
+      // @see https://github.com/storybookjs/storybook/issues/7786
+      useEffect(() => {
+        // Select all elastic textarea elements, run create function on each
+        [...document.querySelectorAll('.js-elastic-textarea')].map(
+          (textarea) => {
+            return createElasticTextArea(textarea);
+          }
+        );
+      }, []);
+      return textDemo({
+        type: select(
+          'type',
+          ['text', 'email', 'search', 'date', 'textarea'],
+          'textarea'
+        ),
+        value: text('value', 'a\nb\nc\nd\ne'),
+        placeholder: text('placeholder', 'Placeholder text…'),
+        rows: number('rows', '2'),
+        id: text('id', 'demo-elastic'),
+        class: text('class', 'js-elastic-textarea'),
+      });
+    }}
+  </Story>
+</Preview>
+
+### JavaScript Usage
+
+Once you have added the `js-elastic-textarea` class to your `textarea` element, you'll also need to add some JavaScript that runs after the page loads, like so:
+
+```js
+// Select all elastic textarea elements, run create function on each
+const elasticTextAreas = [
+  ...document.querySelectorAll('.js-elastic-textarea'),
+].map((textarea) => {
+  return createElasticTextArea(textarea);
+});
```

Comment:
Should this follow our JS prettier/ESLint guidelines for consistency? How do others feel about my question?

Reply:
Prettier is being run on this file. I think ESLint is too

Reply:
@calebeby Oh, I was wondering why the `(textarea) => {}` had parathesis. Usually, we don't add parenthesis when there is only one argument.

Reply:
I _think_ prettier changed this default recently... I don't remember where I read that though.

I think this had to do with it being easier to add more params if the parens are there?

Reply:
https://prettier.io/docs/en/options.html#arrow-function-parentheses

Reply:
Yep prettier 2.0 has parens on all the arrow functions

https://prettier.io/blog/2020/03/21/2.0.0.html#change-default-value-for-arrowparens-to-always-7430httpsgithubcomprettierprettierpull7430-by-kachkaevhttpsgithubcomkachkaev

Reply:
https://twitter.com/ManuelBieh/status/1181880524954050560/photo/1

Reply:
Cool, thanks! 🎓